### ISSUE TITLE
How do I add an item to a sharepoint list?

### ISSUE THREAD

**Issue by dfish88** [vgrem/Office365-REST-Python-Client#241]
I'm trying to add a new item to a sharepoint list that has already been created as follows:

	ctx = ClientContext(site_url).with_credentials(UserCredential(username, password))

	ls = ctx.web.lists.get_by_title(list_name)
	ctx.load(ls)
	ctx.execute_query()

	info = {'Title':'Test','Payroll Name':'Test Name'}
	new_item = ls.add_item(info)
	ctx.execute_query()	

I'm using version 2.1.10.2 and I'm getting this error:

"The property 'Payroll Name' does not exist on type 'SP.Data.ADP_x0020_Employee_x0020_InfoListItem'. Make sure to only use property names that are defined by the type."

Any help would be appreciated.

**Comment by dfish88** [comment]
I've managed to solve the issue that I was having when adding to a sharepoint list. Thought I would put up the solution for others to use.  My scenario is that I'm adding to a sharepoint list that has already been created. I assumed that I would be able to use the column name as it shows up online to add to the list via the API. I've found that this is not the case. For example, the column name "Payroll Name" shows up as "PayrollName" in the API and all the names with spaces have had the spaces removed. I've also found that column names with a colon get encoded. For example, "Work Contact: Work Email" shows up as "WorkContact_x003a_WorkEmail". Another issue I ran into is that long column names get truncated. For example, "Personal Contact: Personal Email" shows up as "PersonalContact_x003a_PersonalEm".  If anyone is having issues like this I would recommend adding an item to the list manually via the sharepoint site and then loading that item in the API and printing the properties using "print(item.properties)" to get the names needed to add and item.
